multidim-interop/js/v0.41/test/ping.spec.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-console */
+/* eslint-env mocha */
+
+import { } from 'aegir/chai'
+// import { createClient } from 'redis'
+
+import { createLibp2p, Libp2pOptions } from 'libp2p'
+// import { tcp } from '@libp2p/tcp'
+import { webSockets } from '@libp2p/websockets'
+import { noise } from '@chainsafe/libp2p-noise'
+import { mplex } from '@libp2p/mplex'
+import { yamux } from '@chainsafe/libp2p-yamux'
+import { multiaddr } from '@multiformats/multiaddr'
+
+describe('ping test', () => {
+  it('should ping', async () => {
+    let nodeFetchImport = "node-fetch"
+    let fetch = (typeof window === "undefined") ?
+      (await import(nodeFetchImport)).default :
+      window.fetch
+
+    const TRANSPORT = process.env.transport
+    const SECURE_CHANNEL = process.env.security
+    const MUXER = process.env.muxer
+    const IS_DIALER_STR = process.env.is_dialer
+    const IP = process.env.ip
+
+    const isDialer = IS_DIALER_STR === 'true'
+    const options: Libp2pOptions = {
+      start: true
+    }
+
+    switch (TRANSPORT) {
+      case 'tcp':
+        // Dynamic so this doesn't get imported in the browser
+        const tcpImport = "@libp2p/tcp"
+        const { tcp } = await import(tcpImport)
+        options.transports = [tcp()]
+        options.addresses = {
+          listen: [`/ip4/${IP}/tcp/0`]
+        }
+        break
+      case 'ws':
+        options.transports = [webSockets()]
+        options.addresses = {
+          listen: isDialer ? [] : [`/ip4/${IP}/tcp/0/ws`]
+        }
+        break
+      default:
+        throw new Error(`Unknown transport: ${TRANSPORT}`)
+    }
+
+    switch (SECURE_CHANNEL) {
+      case 'noise':
+        options.connectionEncryption = [noise()]
+        break
+      default:
+        throw new Error(`Unknown secure channel: ${TRANSPORT}`)
+    }
+
+    switch (MUXER) {
+      case 'mplex':
+        options.streamMuxers = [mplex()]
+        break
+      case 'yamux':
+        options.streamMuxers = [yamux()]
+        break
+      default:
+        throw new Error(`Unknown muxer: ${MUXER}`)
+    }
+
+    const node = await createLibp2p(options)
+
+    try {
+
+      if (isDialer) {
+        const otherMa = process.env.otherMa
+        if (otherMa === undefined) {
+          throw new Error("Failed to wait for listener")
+        }
+        console.log(`node ${node.peerId} pings: ${otherMa}`)
+        const rtt = await node.ping(multiaddr(otherMa))
```

Comment:
this steps fails for me on my own approach due to what I discussed in the slack message. Something with the address being invalid for the transport in the browser. Does this work for you?

Reply:
The browser filters out undialable addresses with: https://github.com/libp2p/js-libp2p-websockets/blob/master/src/filters.ts#L24. So you need a dns + wss multiaddr.

Reply:
You might want to add that to the README, that and any other (aegir) info :) I know readme's aren't added a lot so far, but to me they do help and they can help any potential contributor or reader/user to understand how to use the different tests and related code.

Reply:
Meant more like to a README which would be placed https://github.com/libp2p/test-plans/tree/master/multidim-interop/js/v0.41, usually I like to add readme's or other doc on how to use the code, as I know from experience that not everybody is always an expert in all the different protocols used, and than it can be helpful to see a quick example how to use/run the code. It's next to such a WS Example paragraph that one can then add quick instructions how to run this with a DNS + wss mutliaddr, in all the detail (and nothing more) that is required to be able to run this)).

If this is already well documented, you can next to an example of course just link to that doc with a 1 liner attached to it.

Reply:
> WS Example paragraph that one can then add quick instructions how to run this with a DNS + wss mutliaddr, in all the detail (and nothing more) that is required to be able to run this)).

We can't run this with DNS + WSS yet because of the SSL error. But once we're able to do it I'll make sure to document and subtleties around it.